internal/api/kubernetes/client.go

```diff
@@ -529,74 +533,140 @@ func getAuthorizedClient(
 		return internalClient, nil
 	}
 
+	ra := authv1.ResourceAttributes{
+		Verb:        verb,
+		Group:       gvr.Group,
+		Version:     gvr.Version,
+		Resource:    gvr.Resource,
+		Subresource: subresource,
+		Namespace:   key.Namespace,
+		Name:        key.Name,
+	}
 	if userInfo.Username != "" {
-		// TODO: We are not yet mapping users to Kubernetes SAs, so anyone we could
-		// authenticate currently gets all the same permissions as the Kargo API
-		// server itself.
-		//
-		// We should replace this with a SubjectAccessReview on the SA to determine
-		// if the desired operation is allowed. If it is, we can return the
-		// internalClient.
-		return internalClient, nil
+		serviceAccounts := make(map[string]corev1.ServiceAccount)
+		{
+			list := &corev1.ServiceAccountList{}
+			if err := internalClient.List(ctx, list, libClient.MatchingFields{
+				kubeclient.ServiceAccountsBySubjectIndexField: userInfo.Username,
+			}); err != nil {
```

Comment:
I believe we can constrain this operation by namespace.

If we don't there's a possibility that the user has mappings to a lot of different ServiceAccounts in a lot of different projects, and then we'll end up doing a lot of SubjectAccessReviews.

Also, per #1102, we're only supposed to take into account ServiceAccounts in the project namespace and _then_ fall back on looking for Services accounts in Kargo's own namespace only if no suitable ones are found at the project level.

Reply:
> For example, if the ServiceAccount has a ClusterRole, do we have to deny accessing to the other namespace? 

Not explicitly, no. I don't want to interfere with normal RBAC in any way, of course, but simply defining rules for _how we resolve_ user --> SA mappings does not interfere as far as I can see.

As to what those rules ought to be...

Global mappings in the kargo namespace notwithstanding, it makes sense if project admins have _exclusive_ control over what SA each user is mapped to when interacting with their project. In that case, mappings that other project admins made in other namespaces would be irrelevant.